### ISSUE TITLE
Performance drop (among others: BERT, LSTM_fp32, MobileNET V1 and V2, GRU, ERNIE)

### ISSUE THREAD

**Issue by pmajchrzak** [PaddlePaddle/Paddle#39093]
After this commit: https://github.com/PaddlePaddle/Paddle/commit/5c73a6eaa2be74eaed7e974b433a4c44f6da58b6
a performance drop of several modelas (among others: BERT, LSTM_fp32, MobileNET V1 and V2, GRU, ERNIE) was observed.

It was observed on three different processors: CLX, SKX and CPX. 

### CLX
![image](https://user-images.githubusercontent.com/10847947/150348075-7767de8f-b1b0-4ba3-9f4d-772e44b4fa64.png)

### SKX
![image](https://user-images.githubusercontent.com/10847947/150348167-ab816013-6a31-4563-8865-996297e4e500.png)

### CPX
![image](https://user-images.githubusercontent.com/10847947/150348233-183e816a-0f66-4e36-bfd4-9a4c8ccdbda8.png)

### CMAKE Flags:
``
-DCMAKE_BUILD_TYPE=Release -DWITH_GPU=OFF -DWITH_MKLDNN=ON -DWITH_TESTING=ON -DWITH_PROFILER=OFF -DWITH_STYLE_CHECK=OFF -DON_INFER=ON -DWITH_INFERENCE_API_TEST=ON -DINFERENCE_DEMO_INSTALL_DIR=/data/PaddlePaddle/pp_models -DPY_VERSION=3.6
``
### Environment
```
            OS: Ubuntu18.04 with mkldnn 
 cmake version: 3.10.2
      Compiler: GCC 8
```

for information: @baoachun @jczaja

**Comment by jim19930609** [comment]
Hi pmajchrzak, 
Thanks for this well-formed and detailed feedback. This was a known issue to us and also triggered our internal performance CI. 

We merged a PR yesterday (https://github.com/PaddlePaddle/Paddle/pull/38926) to fix this issue. Can you retry this performance check with develop tip, and see if it still persist? Thanks.

**Comment by pmajchrzak** [comment]
@jim19930609 @baoachun:
> We merged a PR yesterday (#38926) to fix this issue. Can you retry this performance check with develop tip, and see if it still persist? Thanks.
![image](https://user-images.githubusercontent.com/10847947/150966109-fb23dbdd-7cea-46a5-986f-9aab422f1ff0.png)

As you can see in the picture above, there is indeed some increase after the mentioned commit, but basically it only applies to the Bert model. Other models unchanged (see the red, dashed line on the charts - it marks the occurrence of the aforementioned commit - only Bert raised)

But there is another commit, which improves generally more models (https://github.com/PaddlePaddle/Paddle/commit/a56e16a74513c2b1e2d645f002010d3174064d00  and https://github.com/PaddlePaddle/Paddle/commit/43919d0a9eb3abecb1621dc9470ab5dbebd0eb4a):
![image](https://user-images.githubusercontent.com/10847947/150966793-f8a316a1-f2eb-421d-82cc-38efbcc11ec4.png)
